### PR TITLE
feat(legends): legends INNO-630

### PR DESCRIPTION
**Pull request by degliwe**
* upgrade component
* test
* add documentation

### CODE REVIEW

File: framework/components/ecl-forms/ecl-forms-legends/README.md [hunk 1]
```diff
@@ -1 +1,4 @@
-# Legends
```
Comment (emeryro) [inline framework/components/ecl-forms/ecl-forms-legends/README.md]: In other components we kept the global title, so I assume we should keep it here too (and set the other title as second level title)

File: framework/components/ecl-forms/ecl-forms-legends/_legends.scss [hunk 2]
```diff
@@ -1,13 +1,10 @@
...
-Legend
+/**
+* Legend component
+* @define form-legend
 */
 
 $form-item-legend-font-size: map-get($ecl-font-size, 'xs');
 
-.legend {
+.ecl-form-legend {
```
Comment (emeryro) [inline framework/components/ecl-forms/ecl-forms-legends/_legends.scss]: not sure that we have to name class "ecl-form-legend", and not only "ecl-legend". For instance we name other component's classes "ecl-text-input" or "ecl-select"

File: framework/components/ecl-forms/ecl-forms-legends/ecl-forms-legends.twig [hunk 3]
```diff
@@ -1 +1 @@
-<legend>Legend text comes here</legend>
\ No newline at end of file
+<legend class="ecl-form-legend">Legend text comes here</legend>
```
Comment (emeryro) [inline framework/components/ecl-forms/ecl-forms-legends/ecl-forms-legends.twig]: We should put the content into a variable, so we could use composition.

File: framework/components/ecl-forms/ecl-forms-legends/_legends.scss (hunk 2 above)
Comment (degliwe) [inline framework/components/ecl-forms/ecl-forms-legends/_legends.scss]: Ex: we have 2 labels components (ecl-label and ecl-form-label) this legend is specific for form element so it seems more future proof and descriptive to stick to ecl-form-legend.

File: framework/components/ecl-forms/ecl-forms-legends/README.md (hunk 1 above)
Comment (degliwe) [inline framework/components/ecl-forms/ecl-forms-legends/README.md]: Like https://ec-europa.github.io/europa-component-library/components/detail/ecl-buttons ?

Comment (emeryro) [inline framework/components/ecl-forms/ecl-forms-legends/README.md]: Well let's say like every other component, except button :)
We will have to synchronize here

File: framework/components/ecl-forms/ecl-forms-legends/_legends.scss (hunk 2 above)
Comment (emeryro) [inline framework/components/ecl-forms/ecl-forms-legends/_legends.scss]: To be discussed with the team, as we will have this issue again in the future

**Pull request merged**